cicecore/cicedynB/general/ice_forcing.F90

```diff
@@ -4607,8 +4608,12 @@ subroutine hycom_atm_data
       call icepack_query_parameters(Tffresh_out=Tffresh)
       call icepack_query_parameters(secday_out=secday)
 
-      ! current time in HYCOM jday units
-      hcdate = hc_jday(myear,0,0)+ yday+msec/secday
+      ! current time in HYCOM jday units (HYCOM ref year: 1900,12,31,00,00) 
+      hcdate = real(compute_days_between(1900,12,31, &
+                        myear +year_init -1,         &
+                        mmonth+month_init-1,         &
+                        mday  +day_init  -1)         &
+                   ) + (msec  +sec_init)/secday
 
```

Comment:
Is this correct?  myear, mmonth, mday, msec are the model year, month, day, sec at any given time.  They have nothing to do with year_init, month_init, day_init, sec_init.  The _init variables just define the model start time on an initial run.  I think you really want

hcdate = real(compute_days_between(1900,12,31,myear,mmonth,mday)) + msec/secday

?

Reply:
@apcraig I would really like a formula like the one that you provide, which I also started out with.
But unfortunately, it will not lead to the correct answer, because model year,month,day,sec (myear,mmonth,mday,msec) are all *relative* quantities. They are relative to year_init, month_init, day_init, sec_init as I understand. Correct me if I am wrong.

My "HYCOM forcing" is given as days (reals) since 1900-12-31, which is "HYCOM jday" zero.

Example:
target day is yyyymmDD HHMM=20000101 00:00, ie. 1st of January 2000 at time 00:00:
This gives "HYCOM jday"=36160, which is my target. Ie. want hcdate=36160.00000
Same as real(compute_days_between(1900,12,31,2000,01,01))=36160.00000

However:
if year_init=1997, month_init=day_init=1 and sec_init=0, then
  myear=2000-1997=3 , mmonth=1-1=0, mday=1-1=0, msec=0-0=0
which is translated into something like this:
  myear=2, mmonth=12, mday=31, msec=0
and
   real(compute_days_between(1900,12,31,myear,mmonth,mday)) + msec/secday
becomes negative.

Only if year_init=0, month_init=day_init=1 and sec_init=0, the formula will give the correct answer.

Another way could be to store one single public real in ice_calendar (moffset), which relates the model calendar to a real calendar with fixed zero (eg. year_init=0, month_init=day_init=1 and sec_init=0). Them this offset could be used, but it is basically what I am doing in this routine

Then:
hcdate = real(compute_days_between(1900,12,31,myear,mmonth,mday)) + msec/secday + moffset

--o--

If you have another suggestion, I am very open to follow it.

Reply:
In the current version of the time manager, myear, mmonth, mday, and msec are the actual model year, month, day, and sec.  Two of the main points of the revision was to get rid of the "relative" implementation of time and also to get rid of the istep count as a prognostic time variable.  In the current implementation, if it's 2000-01-01 in the model, myear=2000, mmonth=1, mday=1 always.  The _init are only used in namelist to initialize the model time and play no role thereafter.  A big reason to do this was to simplify the implementation and use.  I think this is exactly the kind of place where this comes in.  I think what I suggested before will work.

Reply:
Thanks Tony,
I'll try it once again. Likely a fail 42 from me...

Reply:
Thanks @apcraig
You're right. My purposed equation for hcdate do ONLY work for a specific set of year_init,month_init,day_init, sec_init
The correct hcdate is indeed exactly as you suggested:
```fortran
hcdate = real(compute_days_between(1900,12,31,myear,mmonth,mday)) + msec/secday
```
Code now corrected

--o--

The things that confused me goes back to my restartfile, generated before introducing of the new_ice_calendar:
 `iced.1997-04-01-00000.nc`

Consult restartfile using ncdump:
// global attributes:
		:istep1 = 17609472 ;
		:time = 3037305600. ;
		:time_forc = 3037305600. ;
		:nyr = 97 ;
		:month = 4 ;
		:mday = 1 ;
		:sec = 0 ;

Combined with the following lines in subroutine "init_restart_read" from ice_restart.F90:
```fortran
             status = nf90_get_att(ncid, nf90_global, 'myear', myear)`
            if (status /= nf90_noerr) status = nf90_get_att(ncid, nf90_global, 'nyr', myear)
```
it read `myear=nyr=97`

Adding the global attribute (using ncatted -O -h -a myear,global,a,i,1997 in.nc out.nc) :myear=1997 ;
gives the correct and expected hcdate. Similar, later restart-files looks fine.